### PR TITLE
report for unit tests

### PR DESCRIPTION
**Pull request by ScharfViktor**
part of https://github.com/owncloud/ocis/issues/9730

report for unit tests

result - csv file which includes info about UI test and Unit test

example: 

```
Test-Type,Date,Tests Added,Tests Changed,Tests Deleted,commit-ID
UI Test,2024-08-19T14:30:23+05:45,0,2,0,08d14124cb17b3e928575d67a84c4f71fb0967cb
UI Test,2024-08-19T11:36:04+05:45,0,2,0,9ff8cfabd32b33b22f50d4685f8148f1bbc03336
Unit Test,2024-08-16T11:23:42+02:00,2,0,0,55c32e1b1ad3713bb746ad1f80752d5b589ae269
Unit Test,2024-08-15T12:04:51+02:00,4,7,0,4c3dc5e77b5d01d98c375729793847414c5e900f
Unit Test,2024-08-09T13:48:07+02:00,4,7,0,b4eca2528933406eede42a98a29523606270e88f
Unit Test,2024-08-15T08:10:09+02:00,0,36,0,40abad45909bd2ecb3dd8e137671774422a03862
Unit Test,2024-08-14T07:42:50+02:00,0,36,0,fc82de577f5b973df79dda9a8f29de4573faaa4e
Unit Test,2024-08-13T09:01:54+02:00,0,2,0,aaeb39b0baf4d453a82a73c91d2564681d47e74b
Unit Test,2024-08-12T15:49:52+02:00,0,2,0,94ceef7e16e6a0795e0f9d2915911f991479e554
Unit Test,2024-08-08T14:16:05+02:00,0,2,0,e96da17f1b47e85cf89fc4d575c79503c106773b
Unit Test,2024-08-12T13:02:12+02:00,1,3,0,36cb1c8cc81231897237fedb534bd0fa10095ad2
UI Test,2024-08-12T14:31:06+05:45,2,0,0,d37119c1421b2516c9c145dba7783c461a145421
Unit Test,2024-08-08T15:34:54+02:00,9,6,1,dba04e01b6acb645a23b08471dedb300d76e5807
Unit Test,2024-08-08T15:10:47+02:00,1,0,0,48ab86a5188da11e1d6ae8bfbe3839c821db7ef2
Unit Test,2024-08-08T11:24:58+02:00,8,6,1,503ea48c897e82c92dbb25ba54617ef696f901ba
Unit Test,2024-08-06T15:01:41+02:00,1,0,1,376d3df193eab6ac645b4c853aab62cf7b05f090
Unit Test,2024-08-06T05:07:00+02:00,0,2,0,5b54e5fbe4ac7cb014a14f649b8014abca4a6d5d
UI Test,2024-08-05T13:38:43+02:00,0,2,0,32d81a8f10696981cc7a9bc28d1b047864684c9d
Unit Test,2024-08-05T10:26:41+02:00,10,9,7,d2294cb4e5aaf9e5f931754f3ad8b473cb3df3ed
Unit Test,2024-08-02T13:46:15+02:00,0,2,0,0c36de707b0c6e88ab77491e3b6e87ea119c4409
UI Test,2024-08-02T09:27:25+02:00,1,0,0,4450e210cdd2c3cfd3cac9fa6ac1a09ee693a4b0
```

how to create report: 
```
vscharf:web scharfviktor$ make generate-qa-activity-report MONTH=08 YEAR=2024
pnpm exec node generate-qa-activity-report.js --month 08 --year 2024
CSV report generated successfully. You can find it in", /Users/scharfviktor/Work/web/reports/QA_Activity_Report_08_2024.csv
```

### CODE REVIEW

**Comment by saw-jan** [comment]
>```
>Test-Type,Date,Tests Added,Tests Changed,Tests Deleted,commit-ID
>UI Test,2024-08-19T14:30:23+05:45,0,2,0,08d14124cb17b3e928575d67a84c4f71fb0967cb
>Unit Test,2024-08-16T11:23:42+02:00,2,0,0,55c32e1b1ad3713bb746ad1f80752d5b589ae269
>UI Test,2024-08-19T11:36:04+05:45,0,2,0,9ff8cfabd32b33b22f50d4685f8148f1bbc03336
>Unit Test,2024-08-15T12:04:51+02:00,4,7,0,4c3dc5e77b5d01d98c375729793847414c5e900
>```
Can we group the entries by test type? List all unit test entries first and then UI entries.

File: pnpm-lock.yaml
Comment (saw-jan) [inline pnpm-lock.yaml:1]: is the lock file expected?

**Comment by ScharfViktor** [comment]
> Can we group the entries by test type? List all unit test entries first and then UI entries.

I can, but I'm not sure it's necessary. all data on api unit e2e will be added to csv where they will be sorted.

**Comment by saw-jan** [comment]
>I can, but I'm not sure it's necessary. all data on api unit e2e will be added to csv where they will be sorted.

if the report order in csv file doens't matter then it's fine :+1:

File: pnpm-lock.yaml
Comment (ScharfViktor) [reply pnpm-lock.yaml:1]: yes, this is correct

**Review by tbsbdr — APPROVED** [review]
as far as I can judge: super cool 🙌

**Pull request merged**